src/components/pages/rules/flyouts/ruleStatus/ruleStatus.js

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+import { Observable } from 'rxjs';
+
+import {
+  Btn,
+  FormLabel,
+  ToggleBtn,
+  BtnToolbar,
+  SectionDesc,
+  SummaryCount,
+  SummarySection,
+  AjaxError,
+  Indicator
+} from 'components/shared';
+import { svgs } from 'utilities';
+import { TelemetryService } from 'services';
+import { toNewRuleRequestModel } from 'services/models';
+
+import './ruleStatus.css';
+
+export class RuleStatus extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ruleEntityList: [],
+      status: false,
+      isPending: false,
+      error: undefined,
+    };
+  }
+
+  componentDidMount() {
+    const {rules, rule} = this.props;
+    this.populateRulesState(rules ? rules : [rule]);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {rules, rule} = this.props;
+    this.populateRulesState(rules ? rules : [rule]);
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) this.subscription.unsubscribe();
+  }
+
+  populateRulesState = (rules = []) => {
+    let ruleKey = 1;
+    const ruleEntityList = rules.map(rule => ({
+      rule,
+      key: ruleKey++
```

Comment:
You can use the rule id as a better key that you don't need to compute and is already in the state.